10-angular/resources_and_factories/readme.md

```diff
@@ -1,345 +1,521 @@
+>  
+> **ui.router:** A 3rd party module that functions as a router.  Allows our application to have multiple states.  
+>  
+> **grumbles:** A sub-module to which we attach the components of our application (e.g., controllers).  
+>  
+> **$stateProvider:**  A ui-router service - more on those later! - that allows us to define states in our application.  
+>  
+> **.state:** Used to define an individual state in our application. Arguments include (1) state name and (2) an object that contains information about route, template and controller used.  
+
+#### js/grumbles/grumbles.js
 
-Link to it in index.html
+```js
+"use strict";
 
-```html
-<script src="bower_components/angular-resource/angular-resource.js"></script>
+(function(){
+  angular
+  .module("grumbles", []);
+}());
 ```
+> **grumbles:** Definition of `grumbles` sub-module.  
 
-Add `ngResource` as a dependency to our application.
+#### js/grumbles/index.controller.js
 
 ```js
-// js/app.js
+"use strict";
+
+(function(){
+  angular
+  .module("grumbles")
+  .controller("GrumbleIndexController", [
+    GrumbleIndexControllerFunction
+  ]);
 
-var app = angular.module('grumblr', [
-  'grumbleControllers',
-  'ngResource'
-])
+  function GrumbleIndexControllerFunction(){
+    this.grumbles = [
+      {title: "These"},
+      {title: "Are"},
+      {title: "Hardcoded"},
+      {title: "Grumbles"}
+    ]
+  }
+}());
 ```
+> **GrumbleIndexController:** The name of this controller.  
+>  
+> **GrumbleIndexControllerFunction:** A function that contains this controller's behavior. This is a stylistic decision - we could have passed in an anonymous function to `.controller` if we wanted to.  
 
-create a new file in `js/services/grumble.js` and include in index.html
+You'll notice that, at the moment, we have hard-coded models into the Grumbles controller. Today we'll be learning about `ngResource`, a module that allows us to make calls to that Rails API we set up at the start of class.
 
-```html
-<!-- index.html -->
-<script src="js/services/grumble.js"></script>
-```
+## Factories and Services (15 minutes / 0:40)
 
-```js
-// js/services/grumble.js
-(function() {
-  var grumbleServices = angular.module('grumbleServices', ['ngResource']);
-  grumbleServices.factory('Grumble', ['$resource', function($resource) {
-    return $resource('http://grumblr.wdidc.org/grumbles/:id');
-  }]);
-})();
-```
+First up, we'll convert the hardcoded data to read from an external API using a factory. A factory, however, is not the only way to accomplish this. Let's see what tools we have at our disposal.
 
-and add `grumbleServices` as a dependency in `app.js`
+### Factory
+A factory is an Angular component that adds functionality to an Angular application. Factories allow us to separate concerns and extract functionality that would otherwise be defined in our controller. We do this by creating an object, attaching properties and methods to it and then returning that object. Here's a simple example...
```

Comment:
It's called a factory because it's used to generate new instances of something: grumbles in this case.